net/FlatBuffers/Table.cs

```diff
@@ -112,7 +112,7 @@ public int __vector(int offset)
         // Get the data of a vector whoses offset is stored at "offset" in this object as an
         // ArraySegment&lt;byte&gt;. If the vector is not present in the ByteBuffer,
         // then a null value will be returned.
-        public ArraySegment<byte>? __vector_as_arraysegment(int offset)
+        public ArraySegment<byte>? __vector_as_arraysegment(int offset, int elementSize)
```

Comment:
I guess we're returning `ArraySegment<byte>` instead of `ArraySegment<T>` because of endianness? Though possibly this API would be more useful if we returned `ArraySegment<T>` and asserted we're on little endian? Or is it not possible in C# to make a `ArraySegment<T>` out of bytes?

Reply:
Possibly by some unsafe code tricks but normally you are not allowed to cast no.

The relatively new MemoryMarshal class allows to cast `Span` but curiously not `Memory`, as of yet I fail to understand why. I also have not fully understood if use of MemoryMarshal is as unsafe as unsafe code but https://github.com/dotnet/runtime/issues/41418 indicates it is equivalent to unsafe in many cases.

Reply:
Ok, then this can be merged if @dbaileychess doesn't have any comments.